File: tournaments/eulersTotientFunction/eulersTotientFunction.cpp
int eulersTotientFunction(int n) {
  int divisor = 2,
      result = n;

  while (divisor * divisor <= n) {
    if (n % divisor == 0) {
      while (n % divisor == 0) {
        n /= divisor;
      }
      result -= result / divisor;
    }
    divisor++;
  }
  if (n > 1) {
    result -= result / n;
  }

  return result;
}

File: tournaments/eulersTotientFunction/eulersTotientFunction.js
function eulersTotientFunction(n) {
    let result = n;
    let a = 2;

    while (n >= a * a) {
        if (!(n % a)) {
            while (!(n % a)) n /= a;
            result -= result / a;
        }
        ++a;
    }

    if (n > 1) result -= result / n;

    return result;
}
